specs/latest/2.0/index.html

```diff
@@ -699,12 +694,23 @@ <h2 class="no-toc">Table of contents</h2>
   const GLenum MAX_CLIENT_WAIT_TIMEOUT_WEBGL                 = 0x9247;
 
   /* Buffer objects */
+  void bufferData(GLenum target, BufferDataSource? srcData, GLenum usage,
+                  optional unsigned long srcOffset = 0,
+                  optional unsigned long length = 0);
+  void bufferSubData(GLenum target, GLintptr dstByteOffset,
+                     BufferDataSource srcData,
+                     optional unsigned long srcOffset = 0,
+                     optional unsigned long length = 0);
+
```

Comment:
Assuming the DOM binding generator does what's desired, then I don't think it's necessary to modify the WebGL 1.0 spec, implementation, or most of the tests. Any code in the WebGL 1.0 implementation which currently checks for NULL and generates e.g. INVALID_VALUE errors will simply never be taken for WebGL 2.0 contexts. We might need to change a couple of WebGL 1.0 tests that are run as part of the WebGL 2.0 conformance suite to expect exceptions rather than OpenGL errors for certain illegal inputs.